cmd/zq/zq.go

```diff
@@ -288,3 +302,31 @@ func (c *Command) openOutput() (zbuf.WriteCloser, error) {
 	}
 	return w, nil
 }
+
+func (c *Command) cleanup(f func()) {
+	c.cleanupFns = append(c.cleanupFns, f)
+}
+
+func (c *Command) runCleanup() {
+	for _, fn := range c.cleanupFns {
+		fn()
+	}
+}
+
+func (c *Command) runCPUProfile() {
+	f, err := os.Create(c.cpuprofile)
+	if err != nil {
+		log.Fatal(err)
+	}
+	pprof.StartCPUProfile(f)
+	c.cleanup(pprof.StopCPUProfile)
+}
+
+func (c *Command) runMemProfile() {
+	f, err := os.Create(c.memprofile)
+	if err != nil {
+		log.Fatal(err)
+	}
+	pprof.WriteHeapProfile(f)
```

Comment:
Maybe add a call to runtime.GC as shown at https://pkg.go.dev/runtime/pprof?tab=doc#hdr-Profiling_a_Go_program?

Reply:
Yep, saw this 👍